yjit/src/backend/arm64/mod.rs

```diff
@@ -821,7 +827,11 @@ impl Assembler
                     // the Arm64 assembler works, the register that is going to
                     // be stored is first and the address is second. However in
                     // our IR we have the address first and the register second.
-                    stur(cb, src.into(), dest.into());
+                    match dest.rm_num_bits() {
+                        64 => stur(cb, src.into(), dest.into()),
+                        32 => sturh(cb, src.into(), dest.into()),
```

Comment:
It's odd to that we use a 16 bit store when the size is 32 bit. Does it store the whole word properly when the destination is 32 bit?

Reply:
Oh I didn't realize "halfword" means 16 bits. I guess we should just use `sturh` for 16 and `stur` for 64/32 then.